pkg/http/utils.go

```diff
@@ -494,6 +494,50 @@ func PostRequestBuffer[ResultType any](
 	return result, nil
 }
 
+func PostRequestBufferWithHeaders[ResultType any](
+	options ClientOptions,
+	path string,
+	headers map[string]string,
+	data *bytes.Buffer,
+) (ResultType, error) {
+	var result ResultType
+	client := newRetryClient()
```

Comment:
Is there a way to hoist this up to the module level so that we don't have to create one on every call to `PostRequestBufferWithHeaders`?

Reply:
Yeah, I could see adding a `HTTPClient` with a retry client on it. Then convert this function and the others to methods on it.

Would be a bit of refactoring work, would propose we look at it in a separate PR.